File: semgus-sketch/src/main/java/org/semgus/sketch/Sketch.java
package org.semgus.sketch;

import org.jetbrains.annotations.NotNull;
import org.semgus.java.problem.SemgusProblem;
import org.semgus.pretty.Prettier;
import org.semgus.sketch.base.Stmt;
import org.semgus.sketch.ir.IR;
import org.semgus.sketch.translator.Translator;

public record Sketch(@NotNull Stmt inner) {
  public static @NotNull Sketch fromSemgusProblem(@NotNull SemgusProblem problem, int bnd) {
    var ir = IR.fromSemgusProblem(problem);
    return new Sketch(new Translator(ir, bnd).translate());
  }

  public @NotNull String dump() {
    return Prettier.pretty(inner.codify());
  }
}

File: semgus-sketch/src/main/java/org/semgus/sketch/Sketch.kt
package org.semgus.sketch

import org.semgus.java.problem.SemgusProblem
import org.semgus.pretty.stringPretty
import org.semgus.sketch.base.syntax.Syntax
import org.semgus.sketch.codegen.SyntaxGen
import org.semgus.sketch.translator.toSketchProblem

/**
 * Sketch program.
 */
class Sketch {
  internal lateinit var syntax: Syntax

  fun dump() = syntax.codify().stringPretty()
}

fun fromSemgusProblem(semgusProblem: SemgusProblem, bnd: Int) = Sketch().apply {
  val sketchProblem = semgusProblem.toSketchProblem()
  this.syntax = SyntaxGen(sketchProblem, bnd).gen()
}
